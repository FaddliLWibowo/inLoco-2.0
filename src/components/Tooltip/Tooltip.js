import React from 'react'
import { withContentRect } from 'react-measure'

const calculateDivStyle = (ownHeight, tooltip, scrollTop) => {
    var divStyle = {}
    if (tooltip && tooltip.show) {
        divStyle = {
            left: tooltip.sidebarLeftWidth,
            top: tooltip.top + ((tooltip.parentHeight/2) - (ownHeight/2)) - scrollTop,
        }
    }
    return divStyle
}

const Tooltip = withContentRect('bounds')(({measureRef, measure, contentRect, tooltip, scrollTop}) => {
    var className = ''
    var text = ''
    var title = ''
    if (tooltip && tooltip.show) {
        className = 'tooltip'
        text = tooltip.text === '' ? 'Não tem descrição' : tooltip.text
    }
    return (
        <div  ref={measureRef} className={className} style={calculateDivStyle(contentRect.bounds.height, tooltip, scrollTop)}>
            {text}
        </div>
    )
})

export default Tooltip
