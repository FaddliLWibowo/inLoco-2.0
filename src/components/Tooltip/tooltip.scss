.tooltip {
    max-width: 250px;
    padding: 10px 15px;
    background: $tooltip-bg-color;
    border-radius: 5px;
    box-shadow: 0 8px 6px -6px rgba(47,47,47,.3);
    color: $tooltip-text-color;
    font-size: 12px;
    margin-left: 30px;
    position: absolute;
    z-index: 10000;

    &:before {
        width: 0;
        height: 0;
        border-top: 8px solid transparent;
        border-bottom: 8px solid transparent;
        border-right: 12px solid $tooltip-bg-color;
        content: '';
        margin-top: -8px;
        position: absolute;
        left: -12px;
        top: 50%;
        z-index: 10001;
    }
}
