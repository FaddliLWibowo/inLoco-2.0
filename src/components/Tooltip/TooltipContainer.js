import React from 'react';
import Tooltip from './Tooltip';
import { connect } from 'react-redux';

const mapStateToProps = (state) => {
    return {
        tooltip: state.tooltip
    }
}

const TooltipContainer = connect(
    mapStateToProps,
    null
)(Tooltip)

export default TooltipContainer;
