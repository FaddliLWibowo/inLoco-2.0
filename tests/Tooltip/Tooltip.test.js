import React from 'react';
import Tooltip from '../../src/components/Tooltip/Tooltip.js';
import renderer from 'react-test-renderer';

it('Tooltip component renders correctly with no data', () => {
    const component = shallow(<Tooltip/>);
    expect(component).toMatchSnapshot();
});

it('Tooltip component renders correctly with real data', () => {
    let tooltip = {
        "text":"Índice de Desenvolvimento da Educação Básica (IDEB) da rede estadual de educação. Leitura da tabela: I (Anos Iniciais), F (Anos Finais), E (Estadual), M (Municipal), P (Pública). Fonte: INEP Ano: 2015","show":true,"sidebarLeftWidth":300,"parentHeight":33,"top":230
    }

    const component = shallow(<Tooltip tooltip={tooltip}/>);
    expect(component).toMatchSnapshot();
});
