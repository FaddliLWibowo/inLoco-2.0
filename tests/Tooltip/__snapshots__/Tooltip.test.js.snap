// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders correctly with no data 1`] = `
<Component
  contentRect={
    Object {
      "bounds": Object {},
      "client": Object {},
      "entry": Object {},
      "margin": Object {},
      "offset": Object {},
      "scroll": Object {},
    }
  }
  measure={[Function]}
  measureRef={[Function]}
/>
`;

exports[`Tooltip component renders correctly with real data 1`] = `
<Component
  contentRect={
    Object {
      "bounds": Object {},
      "client": Object {},
      "entry": Object {},
      "margin": Object {},
      "offset": Object {},
      "scroll": Object {},
    }
  }
  measure={[Function]}
  measureRef={[Function]}
  tooltip={
    Object {
      "parentHeight": 33,
      "show": true,
      "sidebarLeftWidth": 300,
      "text": "Índice de Desenvolvimento da Educação Básica (IDEB) da rede estadual de educação. Leitura da tabela: I (Anos Iniciais), F (Anos Finais), E (Estadual), M (Municipal), P (Pública). Fonte: INEP Ano: 2015",
      "top": 230,
    }
  }
/>
`;
